crates/biome_markdown_parser/src/syntax/atx_headings.rs

```diff
@@ -0,0 +1,85 @@
+use crate::parser::MarkdownParser;
+use biome_markdown_syntax::MarkdownSyntaxKind::*;
+use biome_markdown_syntax::T;
+use biome_parser::{
+    prelude::ParsedSyntax::{self, *},
+    Parser,
+    prelude::TokenSource,
+};
+
+pub(crate) fn at_atx_heading(p: &mut MarkdownParser) -> bool {
+    // ATX headings start with 1-6 hash characters
+    if !p.at(T![#]) {
+        return false;
+    }
+
+    // Count consecutive hash characters (max 6)
+    let mut hash_count = 0;
+    while p.nth(hash_count) == T![#] && hash_count < 6 {
+        hash_count += 1;
+    }
+
+    // Must be followed by whitespace or EOL to be a valid heading
+    let next = p.nth(hash_count);
+    next == WHITESPACE || next == NEWLINE || next == T![EOF]
+}
+
+pub(crate) fn parse_atx_heading(p: &mut MarkdownParser) -> ParsedSyntax {
+    let m = p.start();
+
+    // Parse opening hash marks
+    let hash_list_m = p.start();
+    let mut hash_count = 0;
+    while p.at(T![#]) && hash_count < 6 {
+        p.bump(T![#]);
+        hash_count += 1;
+    }
+    hash_list_m.complete(p, MD_HASH_LIST);
+
+    // Skip whitespace after the hash marks
+    if p.at(WHITESPACE) {
+        p.bump(WHITESPACE);
+    }
```

Comment:
IIRC, I think the whitespace is required for it to become a heading. Do you have a source for this behavior?

Reply:
in this example space is skip https://spec.commonmark.org/0.31.2/#example-62
in this example spaces is also skiped https://spec.commonmark.org/0.31.2/#example-62

Reply:
> At least one space or tab is required between the # characters and the heading’s contents, unless the heading is empty. Note that many implementations currently do not require the space.

The code here makes the whitespace optional when it is actually required.